be/src/runtime/data_stream_sender.cpp

```diff
@@ -479,7 +495,30 @@ Status DataStreamSender::send(RuntimeState* state, RowBatch* batch) {
                 hash_val = RawValue::get_hash_value_fvn(
                     partition_val, ctx->root()->type(), hash_val);
             }
-            RETURN_IF_ERROR(_channels[hash_val % num_channels]->add_row(row));
+            auto target_channel_id = hash_val % num_channels;
+            RETURN_IF_ERROR(_channels[target_channel_id]->add_row(row));
+        }
+    } else if (_part_type == TPartitionType::BUCKET_SHFFULE_HASH_PARTITIONED) {
+        // hash-partition batch's rows across channels
+        int num_channels = _channel_shared_ptrs.size();
+
+        for (int i = 0; i < batch->num_rows(); ++i) {
+            TupleRow* row = batch->get_row(i);
+            size_t hash_val = 0;
+
+            for (auto ctx : _partition_expr_ctxs) {
+                void* partition_val = ctx->get_value(row);
+                // We can't use the crc hash function here because it does not result
+                // in uncorrelated hashes with different seeds.  Instead we must use
+                // fvn hash.
+                // TODO: fix crc hash/GetHashValue()
+                //hash_val = RawValue::get_hash_value_fvn(
+                //    partition_val, ctx->root()->type(), hash_val);
+                hash_val = RawValue::zlib_crc32(
```

Comment:
Why comment is fvn hash but code is crc32 hash?

Reply:
This comment is not right， It should be deleted.

Reply:
This comment is not right， I will change this part.